ts/src/connections/tunnelRelayStreamFactory.ts

```diff
@@ -11,14 +11,14 @@ export interface TunnelRelayStreamFactory {
     /**
      * Creates a stream connected to a tunnel relay URI.
      * @param relayUri URI of the tunnel relay to connect to.
-     * @param connectionType Type of stream connection to create, aka websocket sub-protocol.
+     * @param protocols Array of supported connection protocols (websocket sub-protocols).
      * @param accessToken Tunnel host access token, or null if anonymous.
      * @param clientConfig Client config for websocket.
      */
     createRelayStream(
         relayUri: string,
-        connectionType: string,
+        protocols: string[],
         accessToken?: string,
         clientConfig?: IClientConfig,
-    ): Promise<Stream>;
+    ): Promise<{ stream: Stream, protocol: string }>;
```

Comment:
How about declaring an interface for `{ stream: Stream, protocol: string }` ? It would be easier for implementers.

Reply:
It is a `(Stream, string)` tuple in the C# SDK. I don't think this is difficult for implementors, and I prefer to keep them in sync. In fact this kind of pattern is much more common in TypeScript.